pkg/fqdn/rulegen.go

```diff
@@ -260,6 +260,44 @@ func (gen *RuleGen) UpdateGenerateDNS(lookupTime time.Time, updatedDNSIPs map[st
 	return gen.config.AddGeneratedRules(generatedRules)
 }
 
+// ForceGenerateDNS unconditionally regenerates all rules that refer to DNS
+// names in namesToRegen. These are FQDNs and matchPatterns that match them
+// will also cause their rules to regenerate.
+func (gen *RuleGen) ForceGenerateDNS(namesToRegen []string) error {
+	affectedRulesSet := make(map[string]struct{}, 0)
+	for _, dnsName := range namesToRegen {
+		for _, uuid := range gen.sourceRules.LookupValues(dnsName) {
+			affectedRulesSet[uuid] = struct{}{}
+		}
+	}
+
+	// Convert the set to a list
+	var uuidsToUpdate []string
+	for uuid := range affectedRulesSet {
+		uuidsToUpdate = append(uuidsToUpdate, uuid)
+	}
+
+	// Generate a new rule for each sourceRule that needs an update.
+	rulesToUpdate, notFoundUUIDs := gen.GetRulesByUUID(uuidsToUpdate)
+	if len(notFoundUUIDs) != 0 {
+		log.WithField("uuid", strings.Join(notFoundUUIDs, ",")).
+			Debug("Did not find all rules during update")
+	}
+	generatedRules, namesMissingIPs := gen.GenerateRulesFromSources(rulesToUpdate)
+	if len(namesMissingIPs) != 0 {
+		log.WithField(logfields.DNSName, strings.Join(namesMissingIPs, ",")).
+			Warn("Missing IPs for ToFQDN rule")
```

Comment:
This is a warning stating there are missing IPs for ToFQDN, Is this a user problem? If yes, what can the user to to workaround this problem?

Reply:
It is a problem for them, assuming they included a matchName but no IPs are known for it. The only instructions I can think to include would be a little complicated. Something like:
`Missing IPs for toFQDN rule. Ensure DNS data exists via the DNS Proxy or DNS Polling.`
This used to also say: `Using stale data` or something like that. This line is copied from the function above.

Reply:
This is inherited from the DNS poller days, and it is less useful with the proxy. Let's say I include
```
toFQDNS:
  - matchName: foo.com
  - matchName: bar.com
```

If the poller can't resolve these, it means the policy will never work as intended. This warning is the only indication of this (there might also be a timeout error, actually). In the proxy case it is less clear that it's a problem. In that case if a lookup for foo.com happens, bar.com still has no data but that might be ok. In that case, it is more confusing. We now have an api to list the IPs in the DNS cache, so we can check foo.com and bar.com and this is less of an issue.

Are you saying it can be an info? or a debug? I'm ok with either, I guess, since I don't know how to make this reasonable in the dns proxy case.